### PR TITLE
[STY] run style checks on doc

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to https://github.com/nilearn/nilearn/pull/4814#discussion_r1862657286

Changes proposed in this pull request:

- use https://pypi.org/project/doc8/ to run style checks on rst files
  - enforces a 120 line length: mostly to avoid long lines that make PR review difficult 
  - proper naming of code blocks

### CODE REVIEW

**Comment by Remi-Gau** [comment]
wonder if something about this should be mentioned in the contributing doc?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -82,3 +82,10 @@ repos:
         args: [--fix, --show-fixes]
     # Run the formatter.
     -   id: ruff-format
+
+
+-   repo: https://github.com/PyCQA/doc8
+    rev: v1.0.0
+    hooks:
+    -   id: doc8
+        args: [--config=pyproject.toml, nilearn]
```
Comment (man-shu) [inline .pre-commit-config.yaml:91]: So IIUC, would this fix the line violations when committing?

Comment (Remi-Gau) [reply .pre-commit-config.yaml:91]: no doc8 has no auto fix option: you would have to manually fix it

Comment (man-shu) [reply .pre-commit-config.yaml:91]: I see so it would just throw an error while committing

**Review by bthirion — APPROVED** [review]
LGTM, thx

File: doc/maintenance.rst [hunk 2]
```diff
@@ -61,17 +61,18 @@ this, good first issues do not require a separate effort label.
...
+-   |Datasets| This issue is related to datasets or the :mod:`nilearn.datasets` module.
+-   |Discussion| This issue is used to hold a general discussion on a specific topic
+    where community feedback is desired (no need to specify effort, priority, or impact here).
+-   |GLM| This issue is related to the :mod:`nilearn.glm` module.
+-   |Infrastructure| This issue describes a problem with the project's infrastructure (CI/CD...).
+-   |Installation| The issue describes a problem with the installation of Nilearn.
+-   |Maintenance| This issue is related to maintenance work.
+-   |Plotting| The issue is related to plotting functionalities.
+-   |Testing| The issue is related to testing.
+-   |Usage| This issue is a usage question and should have been posted on :neurostars:`neurostars <>`.
```
Comment (emdupre) [inline doc/maintenance.rst:75]: These labels aren't currently rendering in the output : https://output.circle-artifacts.com/output/job/d95b1114-a721-449c-9cc1-b93b8c971245/artifacts/0/dev/maintenance.html

Comment (emdupre) [reply doc/maintenance.rst:75]: weirdly, they loaded after I commented this ! not sure why the long delay -- resolving !

**Pull request merged**